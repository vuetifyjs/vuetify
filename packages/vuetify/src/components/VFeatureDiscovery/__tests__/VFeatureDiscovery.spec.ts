// @ts-nocheck
/* eslint-disable */

// Components
// import VFeatureDiscovery from '../VFeatureDiscovery'
// import VCard from '../../VCard/VCard'
// import VListItem from '../../VList/VListItem'

// Utilities
import {
  mount,
  Wrapper,
} from '@vue/test-utils'
// import { keyCodes } from '../../../util/helpers'
// import { waitAnimationFrame } from '../../../../test'

describe.skip('VFeatureDiscovery.ts', () => {
  type Instance = InstanceType<typeof VFeatureDiscovery>
  let mountFunction: (options?: object) => Wrapper<Instance>

  beforeEach(() => {
    mountFunction = (options = {}) => {
      return mount(VFeatureDiscovery, {
        // https://github.com/vuejs/vue-test-utils/issues/1130
        sync: false,
        ...options,
        mocks: {
          $vuetify: {
            theme: {},
          },
        },
      })
    }
  })

  it('should work', async () => {
    const wrapper = mountFunction({
      propsData: {
        value: false,
        eager: true,
      },
      scopedSlots: {
        activator: '<button v-on="props.on"></button>',
      },
      slots: {
        default: [VCard],
      },
    })

    const activator = wrapper.find('button')
    const input = jest.fn()
    wrapper.vm.$on('input', input)
    activator.trigger('click')

    await wrapper.vm.$nextTick()

    expect(input).toHaveBeenCalledWith(true)
    expect(wrapper.html()).toMatchSnapshot()
    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

})
