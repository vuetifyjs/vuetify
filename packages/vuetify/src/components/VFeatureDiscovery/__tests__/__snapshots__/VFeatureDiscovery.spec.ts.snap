// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VFeatureDiscovery.ts should work 1`] = `
<div class="v-feature-discovery">
  <button>
  </button>
  <div role="feature-discovery"
       class="v-feature-discovery__content theme--light feature-discoverable__content__active "
       style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 8; display: none; z-index: 8;"
  >
    <div class="v-card v-sheet theme--light">
    </div>
  </div>
</div>
`;
